esp/src/eclwatch/XrefQueryWidget.js

```diff
@@ -114,19 +114,20 @@ define([
                     }
                 }, domID);
 
-                retVal.on(".dgrid-row-url:click", function (evt) {
-                    if (context._onRowDblClick) {
-                        var item = retVal.row(evt).data;
-                        context._onRowDblClick(item);
-                    }
-                });
-                retVal.on(".dgrid-row:dblclick", function (evt) {
-                    if (context._onRowDblClick) {
-                        var item = retVal.row(evt).data;
-                        context._onRowDblClick(item);
-                    }
+                ESPUtil.goToPageUserPreference(retVal, "XrefQueryWidget_GridRowsPerPage").then(function(){
+                    retVal.on(".dgrid-row-url:click", function (evt) {
+                        if (context._onRowDblClick) {
+                            var item = retVal.row(evt).data;
+                            context._onRowDblClick(item);
+                        }
+                    });
+                    retVal.on(".dgrid-row:dblclick", function (evt) {
+                        if (context._onRowDblClick) {
+                            var item = retVal.row(evt).data;
+                            context._onRowDblClick(item);
+                        }
+                    });
```

Comment:
There is no need to put the "on" calls in the "then" - its that startup that is important.